extras/wait-for-bricks.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+#Copyright (c) 2018 Red Hat, Inc. <http://www.redhat.com>
+#This file is part of gluster-block.
+#
+#This file is licensed to you under your choice of the GNU Lesser
+#General Public License, version 3 or any later version (LGPLv3 or
+#later), or the GNU General Public License, version 2 (GPLv2), in all
+#cases as published by the Free Software Foundation.
+
+# This file waits for gluster block hosting volume bricks to be up. If all the
+# bricks are up, it exits with success status, otherwise waits for the $timeout
+# seconds before erroring out with the status about
+# online-bricks/expected-online-bricks
+
+LOGDIR="${GB_LOGDIR:-/var/log/gluster-block}"
+
+function printLog()
+{
+    local msg=${1}
+
+    echo "[$(date -u +'%Y-%m-%d %I:%M:%S')] ${msg}" >> "${LOGDIR}/gluster-block-bricks-start.log"
+}
+function volinfo_field()
+{
+    local vol=$1;
+    local field=$2;
+
+    gluster volume info "$vol" --xml | grep -oPm1 "(?<=<$field>)[^<]+"
+}
+
+function is_enabled()
+{
+        case $1 in
+        0|off|no|false|disable)
+                echo "N"
+                ;;
+        1|on|yes|true|enable)
+                echo "Y"
+                ;;
+        *)
+                echo "N/A"
+                ;;
+        esac
+}
+
+function volinfo_option()
+{
+    local vol=$1;
+    local key=$2;
+
+    gluster volume get "$vol" "$key"| awk '{print $NF}' | tail -1
+}
+
+function check_brick_status() {
+        local volname=$1
+        local daemon=$2
+
+        cmd="gluster --xml volume status $volname"
+        if [[ -z $daemon ]]
+         then
+                 $cmd | grep -c '<status>1'
+        else
+                 $cmd | sed -n -e "/${daemon}/,/<status>/ p" | grep -c '<status>1'
+         fi
+}
+
+function volume_online_brick_count()
+{
+        local volname=$1
+        local v1=0
+        local v2=0
+        local v3=0
+        local v4=0
+        local v5=0
+        local tot=0
+
+        #First count total Number of bricks and then subtract daemon status
+        v1=$(check_brick_status "$volname")
+        v2=$(check_brick_status "$volname" "Self-heal")
+        v3=$(check_brick_status "$volname" "Quota")
+        v4=$(check_brick_status "$volname" "Snapshot")
+        v5=$(check_brick_status "$volname" "Tier")
+        tot=$((v1-v2-v3-v4-v5))
+        echo $tot
+}
+
+function volumes_waiting_for_bricks_up()
+{
+        local wait_info
+        local vol_down_info
+        while read -r volname;
+        do
+                if [[ "Started" == $(volinfo_field "$volname" 'statusStr') ]];
```

Comment:
Neither do you need the `;` here. It is only required if you place the `then` on the same line as the `if`.

In case the `$(...)` command fails, the string might be empty and the script can exit with an error. Place the `$(...)` between quotes.